test/facebook_test.exs

```diff
@@ -1,182 +1,598 @@
+          :wow,
+          "#{@test_page_id}_629967087187379",
+          access_token
+        )
+      end
+    end
+  end
 
-    %{"name" => name, "id" => id} = data
-    assert(String.length(name) > 0)
-    assert(id == Integer.to_string(@pageId, 10))
+  describe "object count all" do
+    test "success", %{access_token: access_token} do
+      with_mock :hackney, [
+        request: fn(_method, _url, _headers, _payload, _options) ->
+          {:ok, nil, nil, nil}
+        end,
+        body: fn(_) -> Facebook.GraphMock.object_count_all(:success) end
+      ] do
+        assert {:ok, %{"haha" => haha, "love" => love}} = Facebook.object_count_all(
+          "#{@test_page_id}_629967087187379",
+          access_token
+        )
+
+        assert haha == 135
+        assert love == 10
+      end
+    end
   end
 
-  test "page with fields", context do
-    %{app_access_token: app_access_token} = context
+  describe "long lived access token" do
+    test "success", %{access_token: access_token} do
+      with_mock :hackney, [
+        request: fn(_method, _url, _headers, _payload, _options) ->
+          {:ok, nil, nil, nil}
+        end,
+        body: fn(_) -> Facebook.GraphMock.long_lived_access_token(:success) end
+      ] do
+        assert {:ok, %{
+          "access_token" => access_token,
+          "expires_in" => expires_in,
+          "token_type" => token_type
+        }} = Facebook.long_lived_access_token(
+          @app_id,
+          @app_secret,
+          access_token
+        )
+
+        assert(String.length(access_token) > 0)
+        assert(token_type == "bearer")
+        assert(expires_in > 0)
+      end
+    end
+
+    test "error", %{invalid_access_token: invalid_access_token} do
+      with_mock :hackney, [
+        request: fn(_method, _url, _headers, _payload, _options) ->
+          {:ok, nil, nil, nil}
+        end,
+        body: fn(_) -> Facebook.GraphMock.error() end
+      ] do
+        assert {:error, _} = Facebook.long_lived_access_token(
+          @app_id,
+          @app_secret,
+          invalid_access_token
+        )
+      end
+    end
+  end
 
-    {:json, data} = Facebook.page(@pageId, app_access_token, ["about"])
-    assert(data != nil)
+  describe "new stream" do
+    test "success", %{app_access_token: app_access_token} do
+      with_mock :hackney, [
+        request: fn(_method, _url, _headers, _payload, _options) ->
+          {:ok, nil, nil, nil}
+        end,
+        body: fn(_) -> Facebook.GraphMock.page(:success, :feed) end
+      ] do
+        posts = Facebook.page_feed(
+          :feed,
+          @page_id,
+          app_access_token,
+          1
+        )
+          |> Facebook.Stream.new
+          |> Stream.take(1)
+          |> Enum.to_list
+
+        assert(length(posts) == 1)
+      end
+    end
+  end
+end
 
-    %{"id" => id, "about" => about} = data
-    assert(String.length(about) > 0)
-    assert(id == Integer.to_string(@pageId, 10))
+defmodule Facebook.GraphMock do
```

Comment:
This can be merged into the `test/graph_mock.ex` file

Reply:
So. I might be doing something wrong, but I was getting the following error when trying to use that module. Do you have any idea what I might be doing wrong?

<img width="832" alt="screen shot 2017-10-25 at 9 43 04 pm" src="https://user-images.githubusercontent.com/911914/32035363-8a9cc7d2-b9cd-11e7-9728-ee210d193dd3.png">